pkgs/development/libraries/gstreamer/rs/default.nix

```diff
@@ -0,0 +1,244 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, fetchpatch
+, writeText
+, rustPlatform
+, meson
+, ninja
+, python3
+, pkg-config
+, rust
+, rustc
+, cargo
+, cargo-c
+, nasm
+, gstreamer
+, gst-plugins-base
+, gst-plugins-bad
+, gtk4
+, cairo
+, csound
+, dav1d
+, libsodium
+, libwebp
+, openssl
+, pango
+, Security
+, gst-plugins-good
+, nix-update-script
+# TODO: required for case-insensitivity hack below
+, yq
+, moreutils
+# specify a limited set of plugins to build if not all supported plugins
+, plugins ? null
```

Comment:
```suggestion
, plugins ? [ ]
```

Reply:
Are you suggesting an empty list to mean "all supported plugins"? That's fine if so, right now `null` is used as a special value to signify that it should enable all plugins supported on the given platform

Reply:
null meaning everything is a bit unexpected to me at least. I would expect empty list and null to mean nothing and it is usually easier if a variable has a stricter type expectancy.

Reply:
What would you suggest I use to signify "all supported plugins", then? (e.g. just a string value `"all"`?)

Reply:
(All supported plugins should be good for 99% of users. It's an option at all to accomodate people who are on an uncached platform or cross-compiling to avoid having to build so much just to get one plugin that has no deps aside from `gst-plugins-base`)

Reply:
There is precedent for using `null` for all (e.g. `gimp-with-plugins`)

Reply:
Then we leave it like it is. Should be generally fine.